dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/JdbcIconStore.java

```diff
@@ -0,0 +1,279 @@
+              icon.setKey(rs.getString("iconkey"));
+              icon.setDescription(rs.getString("icondescription"));
+              icon.setCustom(rs.getBoolean("custom"));
+              icon.setKeywords(convertKeywordsJsonIntoSet(rs.getString("keywords")));
+              icon.setCreated(rs.getDate("created"));
+              icon.setLastUpdated(rs.getDate("lastupdated"));
+              icon.setCreatedBy(userService.getUser(rs.getLong("createdby")));
+              icon.setFileResource(fileResourceStore.get(rs.getLong("fileresourceid")));
+              return icon;
+            })
+        .stream()
+        .collect(Collectors.toUnmodifiableSet());
+  }
+
+  private String buildIconQuery(
+      IconOperationParams params, String sql, MapSqlParameterSource parameterSource) {
+    SqlHelper hlp = new SqlHelper(true);
+
+    if (params.hasLastUpdatedStartDate()) {
+      sql += hlp.whereAnd() + " c.lastupdated >= :lastUpdatedStartDate ";
+
+      parameterSource.addValue(
+          ":lastUpdatedStartDate", params.getLastUpdatedStartDate(), Types.TIMESTAMP);
+    }
+
+    if (params.hasLastUpdatedEndDate()) {
+      sql += hlp.whereAnd() + " c.lastupdated <= :lastUpdatedEndDate ";
+
+      parameterSource.addValue(
+          "lastUpdatedEndDate", params.getLastUpdatedEndDate(), Types.TIMESTAMP);
+    }
+
+    if (params.hasCreatedStartDate()) {
+      sql += hlp.whereAnd() + " c.created >= :createdStartDate";
+
+      parameterSource.addValue("createdStartDate", params.getCreatedStartDate(), Types.TIMESTAMP);
+    }
+
+    if (params.hasCreatedEndDate()) {
+      sql += hlp.whereAnd() + " c.created <= :createdEndDate ";
+
+      parameterSource.addValue("createdEndDate", params.getCreatedEndDate(), Types.TIMESTAMP);
+    }
+
+    if (params.hasCustom()) {
+      sql += hlp.whereAnd() + " c.custom = :custom ";
+
+      parameterSource.addValue("custom", params.getCustom(), Types.BOOLEAN);
+    }
+
+    if (params.hasKeywords()) {
+
+      sql += hlp.whereAnd() + " keywords @> cast(:keywords as jsonb) ";
+
+      parameterSource.addValue("keywords", convertKeywordsSetToJsonString(params.getKeywords()));
+    }
+
+    if (params.hasKeys()) {
+      sql += hlp.whereAnd() + " c.iconkey IN (:keys )";
+
+      parameterSource.addValue("keys", params.getKeys());
+    }
+
+    return sql;
+  }
+
+  private String getPaginatedQuery(
+      int page, int pageSize, String sql, MapSqlParameterSource mapSqlParameterSource) {
+
+    sql = sql + " LIMIT :limit OFFSET :offset ";
+
+    int offset = (page - 1) * pageSize;
+
+    mapSqlParameterSource.addValue("limit", pageSize);
+    mapSqlParameterSource.addValue("offset", offset);
+
+    return sql;
+  }
+
+  private Set<String> convertKeywordsJsonIntoSet(String jsonString) {
+    try {
+      return keywordsMapper.readValue(jsonString, new TypeReference<Set<String>>() {});
+    } catch (IOException e) {
+      log.error("Parsing keywords json failed, string value: '{}'", jsonString);
```

Comment:
You can log the exception https://www.baeldung.com/slf4j-log-exceptions

Reply:
fixed